src/simulation/m_acoustic_src.fpp

```diff
@@ -208,6 +210,20 @@ contains
 
             num_points = source_spatials_num_points(ai) ! Use scalar to force firstprivate to prevent GPU bug
 
+            call random_number(phi_rn(1:100))
+            call s_mpi_send_random_number(phi_rn)
```

Comment:
i think i see why you have an mpi routine for this now. you create a new random number every time step, but need all ranks to have the same random number. is that right? if so, then yes it will need its own subroutine like you've already created.

Reply:
Exactly, I will need to broadcast it every time step.